scripts/dataproc-cluster-ui

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env bash
+
+##############################################################
+#  Script to easily open a gcs dataproc cluster's spark UI   #
+##############################################################
+
+#################################
+# Stuff to make and cleanup tmp files
+TMPFILELIST=''
+GCLOUD_PID=''
+function makeTemp()
+{
+  local f
+  f=$( mktemp )
+  TMPFILELIST="${TMPFILELIST} $f"
+  echo $f
+}
+
+function cleanTempVars()
+{
+  rm -f ${TMPFILELIST}
+}
+
+function at_exit()
+{
+  cleanTempVars
+  kill ${GCLOUD_PID} &>/dev/null
+}
+
+trap at_exit EXIT 
+####################################
+
+cluster=$1
+if [[  -z "${cluster}" ]]; then
+	echo ""
+	echo "Usage: $0 <gcs-cluster-name>"
+	echo "Open the spark ui for a dataproc cluster in a new chrome window"
+	exit 1
+fi
+
+port=1080
+
+echo "Connecting to ${cluster} on port ${port}"
+echo ""
+
+tmp=makeTemp
+
+gcloud compute ssh ${cluster}-m -- \
+  -D ${port} -N -n -v 2>${tmp} &
+
+GCLOUD_PID=$!
+
+
+echo "Waiting for SSH connection to be established"
+while ! grep -q "Entering interactive session" ${tmp}; do
```

Comment:
This is kind of sketchy...

Reply:
See also https://stackoverflow.com/questions/1998297/ssh-check-if-a-tunnel-is-alive

Reply:
30s timeout added